pandas/util/_validators.py

```diff
@@ -437,4 +437,4 @@ def validate_insert_loc(loc: int, length: int) -> int:
         loc += length
     if not 0 <= loc <= length:
         raise IndexError(f"loc must be an integer between -{length} and {length}")
-    return loc
+    return loc  # pyright: ignore[reportGeneralTypeIssues]
```

Comment:
Change the sig here too:
```python
def validate_insert_loc(loc: int | np.integer, length: int) -> int | np.integer:
```

Note - if you really don't want to allow `np.integer`, then change the test inside to not use `is_integer()` .